### DISCUSSION TITLE
Check where the message came from and get the attachment

### DISCUSSION THREAD

**Discussion by CatAnonymous** [discord-php/DiscordPHP#1081] [Q&A] [answered]
I have 2 questions:
1. Is it possible to check the user is sending messages from "channel" or "private message"? 
2. Is it possible to get attachments (images, text, audio...) from users? 

 Please give me the solution, thankyou!
```
$discord->on('ready', function (Discord $discord) {
    echo "Bot is online!", PHP_EOL;

    $discord->on(Event::MESSAGE_CREATE, function (Message $message, Discord $discord) {
        echo "{$message->author->username}: {$message->content}", PHP_EOL;
        $message->reply('...');
}
```

**Comment by key2peace** [comment, marked as answer]
1. Private messages have no guild_id
2. when a user uploads an image or audio or text file, they should be available in $message->attachments
3. your example code misses the closing for the $discord->on part, put ``});`` before the closing ``}``

**Reply by CatAnonymous** [reply]
Its work, thanks

```php
        if (count($message->attachments) > 0) {
            var_dump($message->attachments);
        }

        if (null === $message->channel->guild_id) {
          // This part will be processed if it is a Private Message
        }
```
